GCHP 14.6.1 release

yantosca opened the pull request:
### Name and Institution (Required)
Name: Bob Yantosca
Institution: Harvard + GCST

### Describe the update
This PR seeks to merge the release/14.6.1 branch into main in anticipation of the GCHP 14.6.1 release.

The release/14.6.1 branch contains

- Commits from the dev/14.6.1 branch
- Documentation updates from the docs/dev branch.
- Updated version numbers to 14.6.1 in the usual places.

yantosca committed:
geos-chem-shared-docs update: Add doc for S3 bucket listing

This commit informs the GCClassic superproject about the following
commits that were pushed to the geos-chem-shared-docs repository:

31ac061 Tell users to end S3 bucket name with a / to get a listing

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2817 (Consistent GCHP emission diag entries)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

7471dc9cf  Merge PR #2817 (Consistent GCHP emission diag entries)

This PR restores missing emission diagnostic entries for GCHP
(fullchem + carbon simulations).  It also ensures that the
(commented-out) inventory diagnostics are consistent between
fullchem simulations.  For GC in GEOS, the category of the
ALD2_PLANTDECAY emissions has been changed from 3 to 99 to
avoid conflicts with anthropogenic transport emissions.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>
GEOS-Chem update: Merge PR #2817 (Consistent GCHP emission diag entries)

This commit informs the GCClassic superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

7471dc9cf  Merge PR #2817 (Consistent GCHP emission diag entries)

This PR restores missing emission diagnostic entries for GCHP
(fullchem + carbon simulations).  It also ensures that the
(commented-out) inventory diagnostics are consistent between
fullchem simulations.  For GC in GEOS, the category of the
ALD2_PLANTDECAY emissions has been changed from 3 to 99 to
avoid conflicts with anthropogenic transport emissions.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2841 (UVFlux in GCHP alldiags test)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

5b0e7603f  Merge PR #2841 (Restore UVFlux in GCHP alldiags test)

PR #2841 removes a line of code that commented out the UVFlux
diagnostic collection in the GCHP fullchem_alldiags integration
test.  UVFlux has now been restored to this test.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2840 (Fix GFED4 daily frac in ExtData)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

d0c02dd35 Merge PR #2840 (Fix GFED4 daily frac in ExtData.rc)

PR #2840 updates the directory path to the proper GFED4 daily
fraction file in the ExtData.rc (for GCHP only).

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2837 (Fix GEOS-IT cs wind entries)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

eb59d2595 Merge PR #2837 (Fix GEOS-IT cs wind config file entries)

PR #2837 fixes incorrect entries in the run directory template
file geosit.preprocessed_wind_cs.txt (GCHP only).

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2836 (GFAS methanol in ExtData)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

07216fd82 Merge PR #2836 (Add entry for GFAS methanol for ExtData)

PR #2836 adds an entry for methanol from GFAS biomass burning
into the ExtData.rc template file for the GCHP fullchem simulation.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2829 (Fix TOMAS indexing)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

5c7f7e8d6 Merge PR #2829 (Fix TOMAS species indexing #2829)

PR #2829 fixes the indexing for OCIL/OCOB and ECIL/ECOB.  An error
will now be thrown if the ordering of TOMAS species is not as expected.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
geos-chem-shared-docs update: Add link to videos @ BiliBili

This commit informs the GCHP superproject about the following
commits that were pushed to geos-chem-shared-docs:

9e2e1de Update related-docs.rst; Change tables to list-tables

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2847 (Kpp diags in carbon, Hg sims)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

b21eab2ba Merge PR #2847 (KPP diags for carbon, Hg simulatons)

PR #2847 archives the KppDiags, RxnRates, RxnConst HISTORY diagnostics
from the carbon and Hg simulations.  Previously these were only archived
for fullchem simulations.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>GEOS-Chem update: Merge PR #2847 (Kpp diags in carbon, Hg sims)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

b21eab2ba Merge PR #2847 (KPP diags for carbon, Hg simulatons)

PR #2847 archives the KppDiags, RxnRates, RxnConst HISTORY diagnostics
from the carbon and Hg simulations.  Previously these were only archived
for fullchem simulations.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2853 (Obspack I/O fix)

This commit informs the GCClassic superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

e258f7850 Merge PR #2853 (Obspack now looks for "middle" in "time:comment")

PR #2853 adds code to ObsPack so that it will look for either
"middle of the averaging interval" or "midpoint of the averaging interval"
in the "time:comment" field.  This is used to determine whether the
timestamp of the observation should be placed at the start or middle
of the averaging interval.  This fixes the problem raised by @aschuh in
#2849 where ObsPack diagnostic output near the end of a UTC date were
zeroes.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
HEMCO update: Merge PR #325 (Manual InvMEGAN diagnostics)

This commit informs the GCHPsuperproject about the following
commits that were pushed to the GitHub geoschem/HEMCO repository:

2dd30dc Merge PR #325 (Enable manual InvMEGAN diagnostics)

HEMCO PR #282 removed manual InvMEGAN diagnostics from the output.
With this change, it is no longer to possible to get output for these
additional species (such as AAXX, FAXX, and APIN). A quick fix in
PR #325 provides an optional flag to enable the previous behavior.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2862 (Fix miscellaneous minor issues)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

74b0c8e3f Merge PR #2862 (Fix miscellaneous minor issues)

PR #2862 does the following:

1. Wraps several debug print statements for TOMAS in
   "IF ( Input_Opt%Verbose )" blocks.

2. Adds the aod_wavelength_in_nm specification to the GCClassic template
   file geoschem_config.yml.aerosol, for aerosol-only simulations.

3. Adds an error trap in routine Init_Aerosol that will halt a fullchem or
   aerosol-only simulation that does not have at least one AOD wavelength
   specified.

4. Removed unnecessary tags from the photolysis section of
   geoschem_config.yml.aerosol.

5. Added GEOS-Chem Classic run script for the NCAR Derecho cluster.

6. Fixed an sed command so that the KppTime diagnostic will not be included
   in GCClassic "fullchem_alldiags" integration tests.

7. Added the sed command described in point 6 above to the GCHP integration
   test script.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2802 and PR #2869

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

3817b551e Merge PR #2869 (Do not set DELP_DRY=0 if not in restart file)
32b543538 Merge PR #2802 (Add script to generate species tables for RTD doc)

The code for initializing DELP_DRY was moved from hco_interface_gc_mod.F90
to hco_utilities_gc_mod.F90 in PR #2521. This inadvertently led to
negative concentrations in simulations where DELP_DRY was not found in
the restart file. This is because DELP_DRY is updated in the first call
to AirQnt in hco_interfaces_gc_mod.F90. When it is not found in the
restart file, it is reset to zero which causes issues later in the
simulation (manifesting as negative values in mixing). To avoid these
issues, we remove the ELSE statement in PR #2869 setting DELP_DRY to
zero when it's not found in the restart file.


PR #2802 adds the script run/shared/rtd_species_per_simulation.py,
which generates ReST tables of all the species in a given GEOS-Chem
simulation. This can be used to easily update the ReadTheDocs
documentation when new species are added to GEOS-Chem.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GEOS-Chem update: Merge PR #2858 (GEOS-FP proc met w/ C720 fluxes/winds)

This commit informs the GCHP superproject about the following
commits that were pushed to the GitHub geoschem/geos-chem repository:

334bf12bc Merge PR #2858 (GEOS-FP proc. met w/ C720 fluxes and derived winds)

PR #2858 updates GCHP run directory creation to allow GEOS-FP users
to use processed 0.25x0.625 meteorology files with raw C720 meteorology
files.  Previously using GEOS-FP C720 mass fluxes required using raw
GMAO data for ALL meteorology inputs. Now users can choose to use C720
data in advection (mass fluxes, Courant numbers, surface pressure, and
specific humdity) while using processed 0.25x0.3125 files for everything
else. Using raw GMAO input files is still an option but will require
users to download the data from NASA themselves.

PR #2858 also does a general overhaul of the GEOS-FP and GEOS-IT run
directory options for GCHP.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
Merge branch 'dev/no-diff-to-benchmark' into release/14.6.1

This merge brings updates from the "dev/no-diff-to-benchmark"
development stream into the release/14.6.0 branch, in anticipation
of the GCHP 14.6.1 release.

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

Merge branch 'dev/no-diff-to-benchmark' into release/14.6.1

yantosca committed:
Merge 'docs/dev' into release/14.6.1; Update version numbers

This merge brings documentation updates for GCHP 14.6.1 from
the "docs/dev" branch into the release/14.6.1 branch.  Also,
version numbers were updated in the following files:

- CHANGELOG.md
- CMakeLists.txt
- docs/source/conf.py (for ReadTheDocs)

Also docs/source/geos-chem-shared-docs is updated to 9e2e1de

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

lizziel reviewed (changes requested):
Could you also update the changelog to show changes in submodule versions?

yantosca committed:
Added submodule versions to 14.6.1 changelog entry

CHANGELOG.md
- Updated submodule version numbers

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>

yantosca committed:
GCHP 14.6.1 release

This is the official release commit of GCHP 14.6.1.

Updated version numbers in:
- CHANGELOG.md
- CMakeLists.txt
- docs/source/conf.py

Updated GEOS-Chem submodule to version 14.6.1
Updated HEMCO submodule to version 3.11.1
Updated geos-chem-shared-docs submodule to commit 9e2e1de

Signed-off-by: Bob Yantosca <yantosca@seas.harvard.edu>
